test/boost/network_topology_strategy_test.cc

```diff
@@ -936,7 +936,7 @@ SEASTAR_THREAD_TEST_CASE(test_topology_tracks_local_node) {
     BOOST_REQUIRE(!bool(n2->is_this_node()));
     BOOST_REQUIRE_EQUAL(n2->host_id(), host2);
     BOOST_REQUIRE_EQUAL(n2->endpoint(), ip2);
-    BOOST_REQUIRE(n2->dc_rack() == endpoint_dc_rack::default_location);
+    BOOST_REQUIRE(n2->dc_rack() == ip1_dc_rack);
 
```

Comment:
Why do the tests change?

Reply:
Previously, the nodes kept a `endpoint_dc_rack::default_location` that had to be translated using the topology defaults in its config.
Now, since the nodes keep pointers to datacenter and rack, we instantiate the default datacenter and rack when topology is constructed, and we point to them from the node.location, so we get back the read datacenter and rack names instead of the `default_location` placeholder.

Reply:
So, the tests were enshrining incorrect behavior?

Please add that to the changelog. A patch that changes tests is suspect, so it must prove its innocence.

Reply:
sure, will add